Places/Algorithms/ExactOptimalRouteAlgorithm.m

```diff
@@ -0,0 +1,75 @@
+//
+//  ExactOptimalRouteAlgorithm.m
+//  Places
+//
+//  Created by Iris Fu on 8/1/22.
+//
+
+#import "ExactOptimalRouteAlgorithm.h"
+#import "Place.h"
+#import "PlaceTuple.h"
+#import "Route.h"
+
+@implementation ExactOptimalRouteAlgorithm
+
+- (Route *)optimalRouteForPlacesToVisit:(NSArray<Place *> *)places
+                          withValues:(NSDictionary<PlaceTuple *, NSNumber *> *)values {
+    @autoreleasepool {
+        NSArray<NSArray*> *allPossibleRoutes = [self getPermutations:places];
+        NSLog(@"This is all possible routes %@", allPossibleRoutes);
```

Comment:
Too much logging in such a computational intensive algorithm will only slow things down even more. Consider removing the logging or wrapping them within `#if DEBUG` and `#endif` macros so you're only logging in debug builds.